File: app/src/main/java/org/magnum/logger/connectivity/devices/HeadsetHandler.java
package org.magnum.logger.connectivity.devices;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.preference.PreferenceManager;
import android.util.Log;

public class HeadsetHandler extends BroadcastReceiver
{

	private static final String TAG = HeadsetHandler.class.getCanonicalName();

	@Override
	public void onReceive(Context context, Intent intent)
	{
		long time = System.currentTimeMillis();
		try
		{
			int isPlugged = intent.getIntExtra("state", 0);
			if (PreferenceManager.getDefaultSharedPreferences(context).getInt(TAG, -1) != isPlugged)
			{
				new DeviceTable(context).insert(time, intent.getStringExtra("name"), intent.getIntExtra("state", 0));
				PreferenceManager.getDefaultSharedPreferences(context).edit().putInt(TAG, isPlugged).commit();
				Log.d(TAG, "Headset plugged/unplugged at " + time);
			}
		}
		catch (Exception e)
		{
			Log.e(TAG, e.toString());
		}
	}

}

File: app/src/main/java/org/magnum/logger/connectivity/devices/HeadsetHandler.kt
package org.magnum.logger.connectivity.devices

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.util.Log
import androidx.preference.PreferenceManager
import org.magnum.logger.Repository

class HeadsetHandler : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        try {
            val time = System.currentTimeMillis()
            val isPlugged = intent.getIntExtra("state", 0)
            if (PreferenceManager.getDefaultSharedPreferences(context).getInt(TAG, -1) != isPlugged) {
                Repository(context).logDevice(DeviceRecord(time, intent.getStringExtra("name")!!, intent.getIntExtra("state", 0)))
                PreferenceManager.getDefaultSharedPreferences(context).edit().putInt(TAG, isPlugged).apply()
                Log.d(TAG, "Headset plugged/unplugged at $time")
            }
        } catch (e: Exception) {
            Log.e(TAG, e.toString())
        }
    }

    companion object {
        private val TAG = HeadsetHandler::class.java.canonicalName
    }
}